applications/stepswitch/src/stepswitch_resources.erl

```diff
@@ -416,6 +419,16 @@ maybe_resource_to_endpoints(#resrc{id=Id
             maybe_add_proxies(EndpointList, Proxies, Endpoints)
     end.
 
+-spec check_diversion_fields(wapi_offnet_resource:req()) -> ne_binary(). 
+check_diversion_fields(OffnetJObj) ->
+    case wh_json:get_value([<<"Custom-SIP-Headers">>,<<"Diversions">>], OffnetJObj) of
+        'undefined' ->
+            wapi_offnet_resource:outbound_caller_id_number(OffnetJObj);
+        [Diversion|_] ->
+            [_,_,CallerIdNumber,_] = re:split(Diversion,"(:)(\\d*)(@.*)",[{return,binary},{parts,0}]),
```

Comment:
I'd rather you use `binary:split(Diversion, [<<":">>,<<"@">>], ['global'])`.

Also, make sure that you get the right data from JSON by using `wh_json:get_binary_value/2` instead of `get_value/2`